### PR TITLE
[Merged by Bors] - feat: `(s ∩ t).card = s.card + t.card - (s ∪ t).card`

### PR DESCRIPTION
**Pull request by YaelDillies**
once coerced to an `AddGroupWithOne`. Also unify `Finset.card_disjoint_union` and `Finset.card_union_eq`

From LeanAPAP

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/Finset/Card.lean [hunk 1]
```diff
@@ -463,18 +463,22 @@ theorem card_union_le (s t : Finset α) : (s ∪ t).card ≤ s.card + t.card :=
   card_union_add_card_inter s t ▸ Nat.le_add_right _ _
 #align finset.card_union_le Finset.card_union_le
 
-theorem card_union_eq (h : Disjoint s t) : (s ∪ t).card = s.card + t.card := by
+@[simp] lemma card_union_of_disjoint (h : Disjoint s t) : (s ∪ t).card = s.card + t.card := by
```
Comment (sgouezel) [inline Mathlib/Data/Finset/Card.lean:466]: When you rename a theorem, the good practice now is to add a deprecated alias with the old name. Same thing below.

**Comment by Ruben-VandeVelde** [comment]
maintainer merge

File: Mathlib/Data/Finset/Card.lean [hunk 2]
```diff
@@ -463,18 +463,26 @@ theorem card_union_le (s t : Finset α) : (s ∪ t).card ≤ s.card + t.card :=
...
-@[simp]
-theorem card_disjoint_union (h : Disjoint s t) : card (s ∪ t) = s.card + t.card :=
-  card_union_eq h
-#align finset.card_disjoint_union Finset.card_disjoint_union
+-- 2024-02-09
+@[deprecated] alias card_union_eq := card_union_of_disjoint
+@[deprecated] alias card_disjoint_union := card_union_of_disjoint
+
+lemma cast_card_inter [AddGroupWithOne R] :
+    ((s ∩ t).card : R) = s.card + t.card - (s ∪ t).card := by
```
Comment (urkud) [inline Mathlib/Data/Finset/Card.lean:476]: Could you please add a `Nat` version as well? It should be `rw [← card_inter_add_card_union, Nat.add_sub_cancel]`

Comment (urkud) [reply Mathlib/Data/Finset/Card.lean:476]: Otherwise LGTM.
bors d+

**Comment by YaelDillies** [comment]
bors merge

**Comment by acmepjz** [comment]
I think it's good to have `Finset.card_union_eq_iff` which states that `(s ∪ t).card = s.card + t.card` if and only if `Disjoint s t`.

**Pull request closed without merging**